src/cpp/session/modules/SessionDataViewer.R

```diff
@@ -41,11 +41,16 @@
 
 .rs.addFunction("formatDataColumnDispatch", function(col, ...)
 {
-   formatter <- utils::getS3method(
-      "format",
-      class = class(col),
-      optional = TRUE
-   )
+   formatter <- NULL
+   for (class in class(col)) {
+      formatter <- utils::getS3method(
+         "format",
+         class = class,
+         optional = TRUE, 
+         envir = globalenv()
```

Comment:
Older versions of R don't have the `envir` parameter, so I don't think we can use it here.

If it's necessary that methods be looked for in the global environment, I think you could hack it with something like:

```
evalq(utils::getS3method(...), envir = globalenv())
```

Although you might need to `substitute()` in the `class` value to ensure it's found in the right spot.

Reply:
I'll have another look, it might be that they need to be properly registered, as in the test code, or by usual package registration.